test/joystick.js

```diff
@@ -0,0 +1,301 @@
+var MockFirmata = require("./util/mock-firmata"),
+  five = require("../lib/johnny-five.js"),
+  sinon = require("sinon"),
+  Board = five.Board,
+  Joystick = five.Joystick;
+
+var io = new MockFirmata();
+var board = new Board({
+  io: io,
+  debug: false,
+  repl: false
+});
+
+
+exports["Joystick -- Analog"] = {
+
+  setUp: function(done) {
+
+    this.clock = sinon.useFakeTimers();
+    this.analogRead = sinon.spy(MockFirmata.prototype, "analogRead");
+    this.stick = new Joystick({
+      pins: ["A0", "A1"],
+      freq: 100,
+      board: board
+    });
+
+    this.instance = [{
+      name: "hasAxis",
+    }, {
+      name: "x"
+    }, {
+      name: "y"
+    }];
+
+    done();
+  },
+
+  tearDown: function(done) {
+    this.analogRead.restore();
+    this.clock.restore();
+    done();
+  },
+
+  shape: function(test) {
+    test.expect(this.instance.length);
+
+    this.instance.forEach(function(property) {
+      test.notEqual(typeof this.stick[property.name], "undefined");
+    }, this);
+
+    test.done();
+  },
+
+  data: function(test) {
+    var x = this.analogRead.args[0][1];
+    var y = this.analogRead.args[1][1];
+    var spy = sinon.spy();
+
+    test.expect(2);
+
+    this.stick.on("data", spy);
+
+    x(512);
+    y(560);
+
+    this.clock.tick(200);
+
+    test.ok(spy.calledOnce);
+    test.deepEqual(spy.args[0], [{
+      x: 512,
+      y: 560
+    }]);
+
+    test.done();
+  },
+
+  change: function(test) {
```

Comment:
There should be an additional check to ensure the event is not emitted when the value has not changed.